proto/regen/ecocredit/v1alpha1/tx.proto

```diff
@@ -0,0 +1,167 @@
+    bytes metadata = 3;
+}
+
+// MsgCreateClassResponse is the Msg/CreateClass response type.
+message MsgCreateClassResponse {
+
+    // class_id is the unique ID of the newly created credit class.
+    string class_id = 1;
+}
+
+// MsgCreateBatchRequest is the Msg/CreateBatch request type.
+message MsgCreateBatchRequest {
+
+    // issuer is the address of the batch issuer.
+    string issuer = 1;
+
+    // class_id is the unique ID of the class.
+    string class_id = 2;
+
+    // issuance are the credits issued in the batch.
+    repeated BatchIssuance issuance = 3;
+
+    // metadata is any arbitrary metadata to attached to the credit batch.
+    bytes metadata = 4;
+
+    // BatchIssuance represents the issuance of some credits in a batch to a single recipient.
+    message BatchIssuance {
+
+        // recipient is the account of the recipient.
+        string recipient = 1;
+
+        // tradable_units are the units of credits in this issuance that can be traded by this recipient.
+        // Decimal values are acceptable.
+        string tradable_units = 2;
+
+        // retired_units are the units of credits in this issuance that are effectively retired by the issuer on receipt.
+        // Decimal values are acceptable.
+        string retired_units = 3;
+    }
+}
+
+// MsgCreateBatchResponse is the Msg/CreateBatch response type.
+message MsgCreateBatchResponse {
+
+    // batch_denom is the unique ID of the newly created batch.
+    string batch_denom = 1;
+}
+
+// MsgSendRequest is the Msg/Send request type.
+message MsgSendRequest {
+
+    // sender is the address of the account sending credits.
+    string sender = 1;
+
+    // sender is the address of the account receiving credits.
+    string recipient = 2;
+
+    // credits are the credits being sent.
+    repeated SendUnits credits = 3;
+
+    // SendUnits are the tradable and retired units of a credit batch to send.
+    message SendUnits {
+
+        // batch_denom is the unique ID of the credit batch.
+        string batch_denom = 1;
+
+        // tradable_units are the units of credits in this issuance that can be traded by this recipient.
+        // Decimal values are acceptable within the precision returned by Query/Precision.
+        string tradeable_units = 2;
+
+        // retired_units are the units of credits in this issuance that are effectively retired by the issuer on receipt.
+        // Decimal values are acceptable within the precision returned by Query/Precision.
+        string retired_units = 3;
+    }
+}
+
+// MsgSendResponse is the Msg/Send response type.
+message MsgSendResponse { }
+
+// MsgRetireRequest is the Msg/Retire request type.
+message MsgRetireRequest {
+
+    // holder is the address of the holder of the credits.
+    string holder = 1;
+
+    // credits are the credits being retired.
+    repeated RetireUnits credits = 2;
+
+    // RetireUnits are the units of the batch being retired.
```

Comment:
this description should use the same language as the comment in line 134. Also, related to the comment above about changing a vocabulary for units (in proto/regen/ecocredit/v1alpha1/events.proto)